src/promise-withResolvers.ts

```diff
@@ -0,0 +1,30 @@
+/*#__PURE__*/
+export function withResolvers<T>(this: PromiseConstructor) {
+  const out = {} as {
+    promise: Promise<T>
+    resolve: (value: T | PromiseLike<T>) => void
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    reject: (reason?: any) => void
+  }
+  out.promise = new Promise<T>((resolve, reject) => {
```

Comment:
nit - I would `new this` here, but we call this in tests directly, so haven't done so